### PR TITLE
build registry

### PR DESCRIPTION
**Pull request by wy65701436**

### CODE REVIEW

File: make/photon/registry/Dockerfile_Golang [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM golang:1.7.3
```
Comment (reasonerjt) [inline make/photon/registry/Dockerfile_Golang]: This is not a golang image, so let's change the name to Dockerfile.binary

File: make/photon/registry/builder [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+echo "Build registry binary success, then to build photon image..."
+cd $cur
+echo $PHOTONIMAGE
+docker build -f Dockerfile -t $PHOTONIMAGE .
+
+rm -rf $TEMP
+rm -rf binary
+
+echo 'Push image to server.'
```
Comment (reasonerjt) [inline make/photon/registry/builder]: I think it pushed image to docker hub?

File: make/photon/registry/builder [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set +e
+
+echo "Usage: #./builder [registry version] [photon tag] [registry username] [registry password]"
```
Comment (reasonerjt) [inline make/photon/registry/builder]: I think it's docker hub?

File: make/photon/registry/builder [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+cd `dirname $0`
+cur=$PWD
+
+# the temp folder to store distribution source code...
+TEMP=`mktemp -d /$TMPDIR/distribution.XXXXXX`
+git clone -b $VERSION https://github.com/docker/distribution.git $TEMP
+
+echo 'build the registry binary bases on the golang:1.7.3...'
+cp Dockerfile_Golang $TEMP
+docker build -f $TEMP/Dockerfile_Golang -t registry-golang $TEMP
```
Comment (reasonerjt) [inline make/photon/registry/builder]: should it be a docker run or docker build?

Comment (wy65701436) [inline make/photon/registry/builder]: it's should be the docker build.

File: make/photon/registry/builder (hunk 2 above)
Comment (wy65701436) [inline make/photon/registry/builder]: yes, it will push the image to docker hub. What's your suggestion?

Comment (reasonerjt) [inline make/photon/registry/builder]: I'm fine with that, so let's clarify in the message.

File: make/photon/registry/builder (hunk 4 above)
Comment (reasonerjt) [inline make/photon/registry/builder]: This is a little bit, I was thinking we should provide a builder image, and build whatever source via "docker run" put the binary in volume?

**Comment by reasonerjt** [comment]
An overall comment, this PR may not have to be merged in 1.2 timeframe as it does not block anything.

**Pull request merged**